packages/itwinui-css/src/dialog/dialog.scss

```diff
@@ -3,144 +3,105 @@
   }
+
+  &:not(.iui-dialog-visible) {
+    visibility: hidden;
+    opacity: 0;
+  }
+
+  transition: visibility $iui-speed-instant linear, opacity $iui-speed-fast ease-out;
+  transition-delay: $iui-speed-fast, $iui-speed-instant;
+
+  &.iui-dialog-visible {
+    transition-delay: $iui-speed-instant; // remove delay for entry animation
+  }
 }
 
 @mixin iui-dialog-default {
-  .iui-dialog {
-    left: 50%;
-    top: 33%;
-    transform: translate(-50%, -33%);
-    max-width: 50%;
-    min-width: 380px;
-    max-height: 100vh;
-
-    @media screen and (max-width: 400px) {
-      max-width: 95%;
-      width: 95%;
-      min-width: 95%;
-    }
-  }
+  left: 50%;
+  top: 33%;
+  transform: translate(-50%, -33%);
+  max-width: Max(50%, $iui-dialog-width);
+  min-width: $iui-dialog-width;
+  max-height: 100vh;
 }
 
 @mixin iui-dialog-full-page {
-  @media (prefers-reduced-motion: no-preference) {
-    transition: visibility $iui-speed-instant linear $iui-speed-slow, opacity $iui-speed ease-out $iui-speed-fast;
-  }
+  border-radius: 0;
+  height: 100vh;
+  height: 100dvh;
+  width: 100vw;
+  top: 0;
+  left: 0;
+  display: flex;
+  flex-direction: column;
+  will-change: transform;
 
-  .iui-dialog {
-    border-radius: 0;
-    height: 100vh;
-    width: 100vw;
-    top: 0;
-    left: 0;
-    max-width: initial;
-    min-width: initial;
-    display: flex;
-    flex-direction: column;
-    will-change: transform;
-    @media (prefers-reduced-motion: no-preference) {
-      transform: translateY(100%);
-      transition: visibility $iui-speed-instant linear $iui-speed, opacity $iui-speed-instant linear $iui-speed,
-        transform $iui-speed-fast ease-in;
-    }
+  &:not(.iui-dialog-visible) {
+    transform: translateY(100%);
   }
 
-  &.iui-dialog-visible .iui-dialog {
-    transform: translateY(0);
-    @media (prefers-reduced-motion: no-preference) {
+  @media (prefers-reduced-motion: no-preference) {
+    transition: visibility $iui-speed-instant linear $iui-speed, opacity $iui-speed-instant linear $iui-speed,
+      transform $iui-speed-fast ease-in;
+
+    &.iui-dialog-visible {
       transition: transform $iui-speed ease-out;
     }
   }
 }
 
 @mixin iui-dialog-draggable {
-  background-color: transparent;
-  pointer-events: none;
-  z-index: 998;
-
-  .iui-dialog {
-    pointer-events: initial;
-    max-width: 100vw;
-    max-height: 100vh;
-    min-width: 380px;
-    min-height: 144px;
-    display: flex;
-    flex-direction: column;
-    padding: 0;
-    @include themed {
-      border: 1px solid t(iui-color-background-border);
-    }
-  }
+  max-width: 100%;
+  max-height: 100vh;
+  min-width: $iui-dialog-width;
+  min-height: 144px;
+  display: flex;
+  flex-direction: column;
+  border: 1px solid var(--iui-color-background-border);
 }
 
 @mixin iui-dialog-animation {
-  // Full Page Dialog Animations for React
-  &-enter .iui-dialog-full-page .iui-dialog {
+  // TODO: fix react-transition-group for default dialog
```

Comment:
After fixing those, it would be great to also test in React whether animations are still working by linking CSS.